test/val/val_decoration_test.cpp

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Common validation fixtures for unit tests
+
+#include "gmock/gmock.h"
+#include "unit_spirv.h"
+#include "val_fixtures.h"
+#include "source/val/decoration.h"
+
+namespace {
+
+using std::string;
+using std::vector;
+using ::testing::HasSubstr;
+using ::testing::Eq;
+using libspirv::Decoration;
+
+using ValidateDecorations = spvtest::ValidateBase<bool>;
+
+TEST_F(ValidateDecorations, ValidateOpDecorateRegistration) {
+  string spirv = R"(
+    OpCapability Shader
+    OpCapability Linkage
+    OpMemoryModel Logical GLSL450
+    OpDecorate %1 ArrayStride 4
+    OpDecorate %1 Uniform
+    %2 = OpTypeFloat 32
+    %1 = OpTypeRuntimeArray %2
+    ; Since %1 is used first in Decoration, it gets id 1.
+)";
+  const uint32_t id = 1;
+  CompileSuccessfully(spirv);
+  EXPECT_EQ(SPV_SUCCESS, ValidateAndRetrieveValidationState());
+  // Must have 2 decorations.
+  EXPECT_THAT(vstate_->id_decorations(id),
+              Eq(vector<Decoration>{
+                  Decoration(SpvDecorationArrayStride, vector<uint32_t>{4}),
```

Comment:
Did you try this without vector<uint32_t>  annotation?  I think it should work.
(Yup, I just tried GCC.)

The tricky thing is only that you need the vector<Decoration> to be explicit because the Eq template needs a little help to match the vstate_->id_decorations(id) value under test.